Calendar/Parse Handlers/ParseEventHandler.h

```diff
@@ -10,20 +10,18 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
-@protocol ParseEventHandlerDelegate
-
-- (void)successfullyUploadedEvent:(Event *)event;
-- (void)successfullyQueriedWithEvents:(NSMutableArray<Event *> *)events;
-- (void)failedRequestWithMessage:(NSString *)errorMessage;
-
-@end
-
 @interface ParseEventHandler : NSObject
 
-@property (weak, nonatomic) id<ParseEventHandlerDelegate> delegate;
+- (NSString *)getCurrentUsername;
+- (void)queryUserEventsOnDate:(NSDate *)date
+                   completion:(void(^_Nonnull)(NSMutableArray<Event *> * _Nullable events, NSDate *date, NSString * _Nullable error))completion;
 
-- (void)uploadToParseWithEvent:(Event *)newEvent;
-- (NSArray<Event *> *)queryUserEventsOnDate:(NSDate *)date;
+- (void)uploadToParseWithEvent:(Event *)newEvent
+                    completion:(void (^_Nonnull)(Event *event, NSDate *date, NSString * _Nullable error))completion;
+- (void)updateParseObjectWithEvent:(Event *)event
```

Comment:
This is a good example of where we would want to program to interfaces. Let's discuss this more.